Source/WebGPU/WGSL/Lexer.cpp

```diff
@@ -236,6 +236,8 @@ Token Lexer<T>::lex()
                 return makeToken(TokenType::KeywordWorkgroup);
             if (view == "write"_s)
                 return makeToken(TokenType::KeywordWrite);
+            if (view == "array"_s)
```

Comment:
Do we really want to make `array` a keyword? I know the language spec says it is, but why not just treat it as any other type (and say all types can have type parameters, which get matched during type checking)? It would handle `array` and `pointer` systematically instead of as special-cases.

Reply:
That is a choice and it looks like it was Robin's intention.